include/micm/process/process_set.hpp

```diff
@@ -128,10 +141,10 @@ namespace micm
     }
   }
 
-  inline void ProcessSet::AddForcingTerms(
-      const Matrix<double>& rate_constants,
-      const Matrix<double>& state_variables,
-      Matrix<double>& forcing) const
+  template<template<class> typename M>
+    requires(!Vectorizable<M<double>>)
+  inline void
+  ProcessSet::AddForcingTerms(const M<double>& rate_constants, const M<double>& state_variables, M<double>& forcing) const
   {
     // loop over grid cells
     for (std::size_t i_cell = 0; i_cell < state_variables.size(); ++i_cell)
```

Comment:
Thanks. But the exact dimension order should depend on whether we loop **ncells** as the inner- or outermost place?

Reply:
Yes, the existing matrix is ordered like this, with grid cells being the slowest varying index.

The new VectorMatrix has the block-like structure John described, but it still implements the same matrix "policy" so that you can use `Matrix` and `VectorMatrix` interchangeably (primarily for testing), but write optimized functions that loop over the underlying vector in a way that is specific to the VectorMatrix ordering.

Reply:
Thanks for the clarification. That is clear to me now.